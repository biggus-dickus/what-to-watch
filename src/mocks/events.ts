// JSDom has no implementation for common methods of HTML media elements.
// Tests will fail without this. Sad but true.
// https://github.com/jsdom/jsdom/issues/2155#issuecomment-366703395
export const mockHTMLMediaMethods = (): void => {
  window[`HTMLMediaElement`].prototype.load = () => { /* do nothing */ };
  window[`HTMLMediaElement`].prototype.play = () => { /* do nothing */ };
  window[`HTMLMediaElement`].prototype.pause = () => { /* do nothing */ };
};
