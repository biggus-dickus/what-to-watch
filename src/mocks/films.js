export const films = [
  {
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    picUrl: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Bohemian Rhapsody`,
    picUrl: `img/bohemian-rhapsody.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Macbeth`,
    picUrl: `img/macbeth.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Aviator`,
    picUrl: `img/aviator.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `We need to talk about Kevin`,
    picUrl: `img/we-need-to-talk-about-kevin.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `What We Do in the Shadows`,
    picUrl: `img/what-we-do-in-the-shadows.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Revenant`,
    picUrl: `img/revenant.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Johny English`,
    picUrl: `img/johnny-english.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Shutter Island`,
    picUrl: `img/shutter-island.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Pulp Fiction`,
    picUrl: `img/pulp-fiction.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `No Country for Old Men`,
    picUrl: `img/no-country-for-old-men.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Snatch`,
    picUrl: `img/snatch.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Moonrise Kingdom`,
    picUrl: `img/moonrise-kingdom.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Seven Years in Tibet`,
    picUrl: `img/seven-years-in-tibet.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Midnight Special`,
    picUrl: `img/midnight-special.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `War of the Worlds`,
    picUrl: `img/war-of-the-worlds.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Dardjeeling Limited`,
    picUrl: `img/dardjeeling-limited.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Orlando`,
    picUrl: `img/orlando.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
  {
    name: `Mindhunter`,
    picUrl: `img/mindhunter.jpg`,
    pageUrl: `https://thepiratebay.org`
  },
];
