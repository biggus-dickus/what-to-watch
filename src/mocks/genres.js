export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};
