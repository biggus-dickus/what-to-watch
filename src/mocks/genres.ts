import {Genre as GenreType} from '../types'; // eslint-disable-line

export const Genre = {
  ALL: `All genres`,
  COMEDIES: `Comedies`,
  CRIME: `Crime`,
  DOCUMENTARY: `Documentary`,
  DRAMAS: `Dramas`,
  HORROR: `Horror`,
  FAMILY: `Kids & Family`,
  ROMANCE: `Romance`,
  SCI_FI: `Sci-Fi`,
  THRILLERS: `Thrillers`
};

export const mockGenres = Object.values(Genre);

export const mockGenreEntity: GenreType = {
  currentGenre: Genre.CRIME,
  genres: [...mockGenres],
  onGenreChange: jest.fn()
};
