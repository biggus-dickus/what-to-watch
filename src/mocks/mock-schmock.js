export const mockGenres = [`All`, `BDSM`, `Gay gymnastics`, `Psycho-dramas for aestheticizing faggots`];

export const mockFilms = [
  {
    name: `L'elephant vert`,
    genre: [mockGenres[1], mockGenres[2]],
    pageUrl: `https://www.imdb.com/title/tt1003080/`,
    picUrl: `http://lurkmore.so/images/4/47/Slonik_macro.jpg`,
    videoUrl: `https://storage.googleapis.com/coverr-main/mp4%2FCam-Man.mp4`
  },
  {
    name: `Sal√≤ o le 120 giornate di Sodoma`,
    genre: [mockGenres[3]],
    pageUrl: `https://www.imdb.com/title/tt0073650/`,
    picUrl: `https://m.media-amazon.com/images/M/MV5BYjE0NTU3MTUtYjE0OC00ZGM1LWIwNzYtNDU2YWI2YWJjZjEwXkEyXkFqcGdeQXVyMTA2ODMzMDU@._V1_SY1000_CR0,0,764,1000_AL_.jpg`,
    videoUrl: `etc/na-zdorovnya.mp4`
  },
  {
    name: `Un Chien Andalou`,
    genre: [mockGenres[3]],
    pageUrl: `https://www.imdb.com/title/tt0020530/`,
    picUrl: `https://m.media-amazon.com/images/M/MV5BMjZlY2RjM2ItZGEwOC00ODQwLWFhZDYtMjE3ODMyYWNiOWNlXkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SY1000_CR0,0,706,1000_AL_.jpg`,
    videoUrl: `etc/na-zdorovnya.mp4`
  }
];
