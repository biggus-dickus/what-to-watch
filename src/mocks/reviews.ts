import {Review} from '../types'; // eslint-disable-line

export const mockReviews: Review[] = [
  {
    id: 1,
    user: {id: 19, name: `Christina`},
    rating: 4,
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-06-17T16:39:57.295Z`
  }, {
    id: 2,
    user: {id: 14, name: `Corey`},
    rating: 3,
    comment: `Unfortunately we don't have a reliable way to tell the true ratings of a movie.`,
    date: `2019-06-20T16:39:57.295Z`
  }, {
    id: 3,
    user: {id: 13, name: `Zak`},
    rating: 4,
    comment: `I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.`,
    date: `2019-06-12T16:39:57.295Z`
  }
];
