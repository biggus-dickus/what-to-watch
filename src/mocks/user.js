const mockUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatar_url: `img/1.png` // eslint-disable-line
};

export default mockUser;
