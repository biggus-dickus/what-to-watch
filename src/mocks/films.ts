import {Film} from '../types'; // eslint-disable-line
import {Genre} from './genres';

export const mockFilms: Film[] = [
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Fantastic Beasts: The Crimes of Grindelwald`,
    id: 1,
    genre: Genre.FAMILY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    bgColor: `green`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 7.77,
    released: 2019,
    runTime: 120,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Bohemian Rhapsody`,
    id: 2,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/bohemian-rhapsody.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    bgColor: `blue`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1654,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Macbeth`,
    id: 3,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/macbeth.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4%2FCam-Man.mp4`
  },
  {
    bgColor: `rebeccapurple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 5.55,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Aviator`,
    id: 4,
    genre: Genre.ROMANCE,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/aviator.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `We need to talk about Kevin`,
    id: 5,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/we-need-to-talk-about-kevin.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/5/5b/Marnie_%281964%29_trailer.webm/Marnie_%281964%29_trailer.webm.360p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `What We Do in the Shadows`,
    id: 6,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/what-we-do-in-the-shadows.jpg`,
    previewVideo: `https://upload.wikimedia.org/wikipedia/commons/transcoded/2/23/Dr_Strangelove_-_Official_Trailer.webm/Dr_Strangelove_-_Official_Trailer.webm.480p.vp9.webm`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Revenant`,
    id: 7,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/revenant.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Harvesting%20Rice.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.10,
    released: 2005,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Johny English`,
    id: 8,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/johnny-english.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Lake%20Day.mp4`
  },
  {
    bgColor: `pink`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 2009,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Shutter Island`,
    id: 9,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/shutter-island.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Deserted%20Island.mp4`
  },
  {
    bgColor: `purple`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Quentin Tarantino`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 9.50,
    released: 1994,
    runTime: 180,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Pulp Fiction`,
    id: 10,
    genre: Genre.CRIME,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/pulp-fiction.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stunning%20Mountains.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `No Country for Old Men`,
    id: 11,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/no-country-for-old-men.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Gubat%20Beach.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Guy Ritchie`,
    isFavourite: true,
    posterImage: `poster.jpg`,
    rating: 8.36,
    released: 2004,
    runTime: 125,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Snatch`,
    id: 12,
    genre: Genre.COMEDIES,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/snatch.jpg`,
    previewVideo: `etc/na-zdorovnya.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Moonrise Kingdom`,
    id: 13,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/moonrise-kingdom.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Caramoan%20Drone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Seven Years in Tibet`,
    id: 14,
    genre: Genre.DOCUMENTARY,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/seven-years-in-tibet.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Ants%20Walking.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Midnight Special`,
    id: 15,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/midnight-special.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Boat%20in%20Caramoan.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `War of the Worlds`,
    id: 16,
    genre: Genre.SCI_FI,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/war-of-the-worlds.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Blue%20Joy.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Dardjeeling Limited`,
    id: 17,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/dardjeeling-limited.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Stop-Sign.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Orlando`,
    id: 18,
    genre: Genre.DRAMAS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/orlando.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/Steaming-Yellowstone.mp4`
  },
  {
    bgColor: `red`,
    bgImage: `pic.png`,
    description: `Come Tatyana with a duck, we will eat and we will fuck`,
    director: `Alan Smithie`,
    isFavourite: false,
    posterImage: `poster.jpg`,
    rating: 6.66,
    released: 1984,
    runTime: 100,
    scoresCount: 100500,
    starring: [`Ben Dover`, `Major Woodie`, `Private Parts`, `I. C. Wiener`],
    video: `https://thepiratebay.org`,
    name: `Mindhunter`,
    id: 19,
    genre: Genre.THRILLERS,
    pageUrl: `https://thepiratebay.org`,
    previewImage: `img/mindhunter.jpg`,
    previewVideo: `https://storage.googleapis.com/coverr-main/mp4/California-Desert.mp4`
  }
];
