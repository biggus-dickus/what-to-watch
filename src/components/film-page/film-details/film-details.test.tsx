import * as React from 'react';
import * as renderer from 'react-test-renderer';
import * as Adapter from 'enzyme-adapter-react-16';
import {configure, shallow} from 'enzyme';

import {mockFilms} from '../../../mocks/films';
import FilmDetails from './film-details';

configure({adapter: new Adapter()});


const props = {
  director: mockFilms[0].director,
  genre: mockFilms[0].genre,
  released: mockFilms[0].released,
  runTime: mockFilms[0].runTime,
  starring: mockFilms[0].starring
};

describe(`FilmDetails test suite`, () => {
  it(`<FilmDetails /> should render correctly`, () => {
    expect(renderer.create(<FilmDetails {...props} />).toJSON()).toMatchSnapshot();
  });

  it(`should correctly transform runTime in minutes into a hh:mm format`, () => {
    const wrapper = shallow(<FilmDetails {...props} />);
    const node = wrapper.find(`[data-test="at-film-duration"]`);

    expect(node.text()).toEqual(`1h 40m`);
  });
});
