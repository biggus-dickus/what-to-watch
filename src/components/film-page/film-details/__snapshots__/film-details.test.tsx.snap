// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmDetails test suite <FilmDetails /> should render correctly 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        Alan Smithie
      </span>
    </p>
    <div
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <ul
        className="movie-card__details-value"
      >
        <li>
          Ben Dover
          ,
        </li>
        <li>
          Major Woodie
          ,
        </li>
        <li>
          Private Parts
          ,
        </li>
        <li>
          I. C. Wiener
        </li>
      </ul>
    </div>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
        data-test="at-film-duration"
      >
        1
        h 
        40
        m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Kids & Family
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        1984
      </span>
    </p>
  </div>
</div>
`;
